tests/ops/integration_tests/test_integration_erasure_order.py

```diff
@@ -0,0 +1,350 @@
+        min_size=1,
+        keys=["refunds_to_orders_id"],
+    )
+
+    temp_masking = CONFIG.execution.masking_strict
+    CONFIG.execution.masking_strict = False
+
+    with pytest.raises(TraversalError) as exc:
+        await graph_task.run_erasure(
+            privacy_request,
+            erasure_policy_complete_mask,
+            graph,
+            [saas_erasure_order_connection_config],
+            identity_kwargs,
+            get_cached_data_for_erasures(privacy_request.id),
+            db,
+        )
+
+    assert (
+        f"The values for the `erase_after` fields caused a cycle in the following collections"
+        in str(exc.value)
+    )
+
+    CONFIG.execution.masking_strict = temp_masking
+
+
+@pytest.mark.integration_saas
+@pytest.mark.asyncio
+@mock.patch("fides.api.ops.service.connectors.saas_connector.SaaSConnector.mask_data")
+async def test_saas_erasure_order_request_task_resume_from_error(
+    mock_mask_data,
+    db,
+    policy,
+    erasure_policy_complete_mask,
+    saas_erasure_order_connection_config,
+    saas_erasure_order_dataset_config,
+) -> None:
+    privacy_request = PrivacyRequest(
+        id=f"test_saas_erasure_order_request_task_resume_from_error_{random.randint(0, 1000)}"
+    )
+    identity_attribute = "email"
+    identity_value = "test@ethyca.com"
+    identity_kwargs = {identity_attribute: identity_value}
+    identity = Identity(**identity_kwargs)
+    privacy_request.cache_identity(identity)
+
+    dataset_name = saas_erasure_order_connection_config.get_saas_config().fides_key
+    merged_graph = saas_erasure_order_dataset_config.get_graph()
+    graph = DatasetGraph(merged_graph)
+
+    v = await graph_task.run_access_request(
+        privacy_request,
+        policy,
+        graph,
+        [saas_erasure_order_connection_config],
+        {"email": "test@ethyca.com"},
+        db,
+    )
+
+    assert_rows_match(v[f"{dataset_name}:orders"], min_size=1, keys=["orders_id"])
+    assert_rows_match(v[f"{dataset_name}:refunds"], min_size=1, keys=["refunds_id"])
+    assert_rows_match(v[f"{dataset_name}:labels"], min_size=1, keys=["labels_id"])
+    assert_rows_match(v[f"{dataset_name}:products"], min_size=1, keys=["products_id"])
+    assert_rows_match(
+        v[f"{dataset_name}:orders_to_refunds"],
+        min_size=1,
+        keys=["orders_to_refunds_id"],
+    )
+    assert_rows_match(
+        v[f"{dataset_name}:refunds_to_orders"],
+        min_size=1,
+        keys=["refunds_to_orders_id"],
+    )
+
+    temp_masking = CONFIG.execution.masking_strict
+    CONFIG.execution.masking_strict = False
+
+    # mock the mask_data function so we can force an exception on the "refunds_to_orders"
+    # collection to simulate resuming from error
+    def side_effect(node, policy, privacy_request, rows, input_data):
+        if node.address.collection == "refunds_to_orders":
+            raise Exception("Error executing refunds_to_orders task")
+        return 1
```

Comment:
Decided to keep the "resume from error" test here and used `side_effect` to simulate an error and the subsequent fix